### ISSUE TITLE
Feature Request: OpenWRT support

### ISSUE THREAD

**Issue by duoi** [gabe565/CastSponsorSkip#35]
Firstly I’d like to thank the creator and contributors for this fantastic product.

I have an OpenWRT based router and I like the idea of being to install this directly on there. I’m not sure what the complications are, but if it is a standalone binary, then it might just need to be packaged in the opkg format.

**Comment by gabe565** [comment]
Hi! CastSponsorSkip does get built to a standalone binary, so it should work on your device. Unfortunately, the packaging tool I use (called GoReleaser) doesn't seem to support the opkg format, but you could still download the binary to your device! Do you know which CPU architecture your router uses?

**Comment by duoi** [comment]
That's cool, I'll see if I can roll something for it. It is MIPS-based however.

**Comment by gabe565** [comment]
I'll search some more to see if there's any way for me to build an opkg.

I suspected it was MIPS. Go supports MIPS compilation so I'll go ahead and enable that even if I can't build an opkg!

**Comment by gabe565** [comment]
Interesting. [This thread](https://forum.openwrt.org/t/install-deb-package/30831/4) makes it seem like opkg can install deb files, so I wonder if I just need to turn on MIPS compilation...

**Comment by duoi** [comment]
That would be really interesting, I didn't know that.

Digging into this further, it looks like I'm out of space on this router so wouldn't be able to go down the `opkg` route either way, and I'd end up just having some script run on startup to pull the binary down onto `/tmp`.

If you are able to make it work via `opkg` though I think it would be incredible. It negates the need for a separate server entirely.

**Comment by duoi** [comment]
Hi @gabe565, hope you are well! Do you plan on turning on MIPS compilation any time soon? Very happy to help test it!

**Comment by atesin** [comment]
i dont believe you could ever could be able to install a debian package inside an openwrt system... the package system, the architecture, everything are lights ahead different

the only reference to sponsorblock i found in openwrt is this: https://forum.openwrt.org/t/youtube-adblock-does-it-exist-on-openwrt-platform/184122/128
